### PR TITLE
config: new "reverse" interpolation function

### PR DESCRIPTION
**Pull request by sargun**
It reverses a list.

### CODE REVIEW

**Comment by sargun** [comment]
First PR on Terraform. This is useful for `https://www.terraform.io/docs/providers/aws/d/ami_ids.html`, where it says:

> ids is set to the list of AMI IDs, sorted by creation time in descending order.

Assuming you want it in ascending order, you need to reverse the list. for_each may solve this, but for existing terraform templates, this is invaluable.

**Comment by apparentlymart** [comment]
Hi @sargun! Thanks for working on this.

The implementation here looks good to me. The interpolation functions are frozen for the moment while we complete some refactoring in preparation for the 0.12 release, so we'll need to hold on merging this for a little while, but I've labelled this so we'll find it again during our 0.12 release preparation.

The way functions are defined will be changing once we complete this refactoring, so this PR will need some changes to adapt to the new style but we'd be happy to do that step at merge time since it'll just be the same mechanical updates we'll be doing for a number of other pending function pull requests.

Thanks again! We'll be back here to merge this as we get closer to the 0.12 release.

**Comment by sargun** [comment]
@apparentlymart Sounds good. Thanks for taking a look at it. Is there a rough timeline as to 0.12 yet?

File: config/interpolate_funcs.go [hunk 1]
```diff
@@ -941,6 +942,25 @@ func interpolationFuncReplace() ast.Function {
...
+func interpolationFuncReverse() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeList},
+		ReturnType: ast.TypeList,
+		Variadic:   false,
+		Callback: func(args []interface{}) (interface{}, error) {
+			inputList := args[0].([]ast.Variable)
+
+			reversedList := make([]ast.Variable, len(inputList))
+			for idx := range inputList {
```
Comment (theodesp) [inline config/interpolate_funcs.go]: One could say that you could iterate over `len(inputList) /2` and do swaps:
example pseudocode
```
        r := make([]int, len(inputList))
	copy(r, inputList)
	for i, j := 0, len(r)-1; i < len(r)/2; i, j = i+1, j-1 {
		r[i], r[j] = r[j], r[i]
	}
	return r, nil
```

**Comment by apparentlymart** [comment]
Sorry for the delay here, @sargun! I rebased your commit to latest master and then ported it into the new functions mechanism in 5922f5a1fb6f5c2789d5771d08dc89d66dca49a7. This'll be included in the forthcoming v0.12.0-beta2 release.

**Pull request merged**